core/trino-spi/src/test/java/io/trino/spi/TestSpiBackwardCompatibility.java

```diff
@@ -126,33 +126,28 @@ private static void addClassEntities(ImmutableSet.Builder<String> entities, Clas
         if (!includeDeprecated && clazz.isAnnotationPresent(Deprecated.class)) {
             return;
         }
+
         entities.add("Class: " + clazz.toGenericString());
         for (Class<?> nestedClass : clazz.getDeclaredClasses()) {
             addClassEntities(entities, nestedClass, includeDeprecated);
         }
+
         for (Constructor<?> constructor : clazz.getConstructors()) {
-            if (!includeDeprecated && constructor.isAnnotationPresent(Deprecated.class)) {
-                continue;
+            if (includeDeprecated || !constructor.isAnnotationPresent(Deprecated.class)) {
+                entities.add("Constructor: " + constructor.toGenericString());
             }
-            entities.add("Constructor: " + constructor.toGenericString());
         }
-        for (Method method : clazz.getDeclaredMethods()) {
-            if (!isPublic(method.getModifiers())) {
-                continue;
-            }
-            if (!includeDeprecated && method.isAnnotationPresent(Deprecated.class)) {
-                continue;
+
+        for (Method method : clazz.getMethods()) {
```

Comment:
The code was asking for `getDeclaredMethods` so that we don't include inherited members, which `getMethods` does.